msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenMockTest.java

```diff
@@ -0,0 +1,577 @@
+                        .build();
+
+                publicClientApplication.acquireTokenSilentAsync(silentParameters);
+                RoboTestUtils.flushScheduler();
+            }
+
+        }.performTest();
+    }
+
+    @Test
+    public void forceRefreshWorks() {
+        new AcquireTokenBaseTest() {
+
+            @Override
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+                final IAccount account = loadAccountForTest(publicClientApplication);
+                final AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+                        .withScopes(Arrays.asList(SCOPES))
+                        .forceRefresh(true)
+                        .forAccount(account)
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                publicClientApplication.acquireTokenSilentAsync(silentParameters);
+                RoboTestUtils.flushScheduler();
+            }
+
+        }.performTest();
+    }
+
+    @Test
+    public void silentCallFailsIfAccountNotProvided() {
+        new AcquireTokenBaseTest() {
+
+            @Override
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+
+                final AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+                        .withScopes(Arrays.asList(SCOPES))
+                        .forceRefresh(false)
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                fail("Unexpected success");
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                Assert.assertTrue(exception instanceof MsalClientException);
```

Comment:
nit: verify that we're getting the expected error message. Here and every test case we're expecting an exception.

Reply:
Why do you think we should check for the message? I think this can cause problems because error messages may change from time to time and that would require us to refactor the test whenever our error message changes. I think checking for type of exception should be good enough. What do you say?

Reply:
Actually, error code / error message should not be changed. (It's a contract between us and developers, and any changes made will not be obvious to them, as they should still be able to build and might not be aware of it at all).

Making a check here will help us catch unexpected changes.

Reply:
yes actual error code from the server should not change. So validating for error code would be good and should not cause any harm, instead it will validate server behavior